[openwrt-24.10] rockchip: Add support for RK3568 LinkEase EasePi R1

jjm2473 opened the pull request:
This PR is a backport of https://github.com/openwrt/openwrt/pull/20147 https://github.com/openwrt/openwrt/pull/20779 and https://github.com/openwrt/openwrt/pull/20772 .

Note: #20772 has been slightly modified to be compatible with u-boot 2024.10.

---------------------

Specification:
- Rockchip RK3568
- 2GB/4GB LPDDR4 RAM
- 16GB on-board eMMC
- 1x M.2 key for 2280 NVMe (PCIe 3.0)
- 1x USB 3.0 Type-A
- 1x USB 2.0 Type-C (for USB flashing)
- 2x 1000 Base-T (native, RTL8211F)
- 2x 2500 Base-T (PCIe, RTL8125B)
- 1x HDMI 2.0 Output
- 12v DC Jack
- 1x Power key connected to PMIC
- 2x LEDs (one static power supplied, one GPIO controlled)

Debug Serial Port: 3.3V TTL, 3-pin 2.54mm pitch connector, 1500000 bauds,
'J7' on board with G/R/T (GND/RX/TX) pins marked

Installation - eMMC:
-Boot official firmware (OpenWRT based)
-Upload sysupgrade.img.gz to /tmp/firmware.bin (with scp or web page)
-Flash to eMMC: sysupgrade -n -p -F /tmp/firmware.bin

jjm2473 committed:
uboot-rockchip: add support for LinkEase EasePi R1

generic-rk3568 is enough

Signed-off-by: Liangbin Lian <jjm2473@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20147
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit ef093299dbb8d9e9b8120be9d5314dbf4af1528d)

jjm2473 committed:
rockchip: Add support for RK3568 LinkEase EasePi R1

Specification:
- Rockchip RK3568
- 2GB/4GB LPDDR4 RAM
- 16GB on-board eMMC
- 1x M.2 key for 2280 NVMe (PCIe 3.0)
- 1x USB 3.0 Type-A
- 1x USB 2.0 Type-C (for USB flashing)
- 2x 1000 Base-T (native, RTL8211F)
- 2x 2500 Base-T (PCIe, RTL8125B)
- 1x HDMI 2.0 Output
- 12v DC Jack
- 1x Power key connected to PMIC
- 2x LEDs (one static power supplied, one GPIO controlled)

Debug Serial Port: 3.3V TTL, 3-pin 2.54mm pitch connector, 1500000 bauds,
'J7' on board with G/R/T (GND/RX/TX) pins marked

Installation - eMMC:
-Boot official firmware (OpenWRT based)
-Upload sysupgrade.img.gz to /tmp/firmware.bin (with scp or web page)
-Flash to eMMC: sysupgrade -n -p -F /tmp/firmware.bin

Signed-off-by: Liangbin Lian <jjm2473@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20147
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit 8ca4caacd039bba6d53d3c578fbcf21b6a2fbff2)
[refresh patches upon 24.10 branch]

jjm2473 commented:
I have tested it on my own R1.
Most of the time there are no problems.
```
DDR 2d653b3476 typ 24/01/20-15:04:19,fwver: v1.21
In
LP4/4x derate en, other dram:1x trefi
ddrconfig:7
LPDDR4, 324MHz
BW=32 Col=10 Bk=8 CS0 Row=16 CS1 Row=16 CS=2 Die BW=16 Size=4096MB
tdqss: cs0 dqs0: 48ps, dqs1: -72ps, dqs2: -24ps, dqs3: -144ps, 
tdqss: cs1 dqs0: 48ps, dqs1: -72ps, dqs2: -24ps, dqs3: -120ps, 

change to: 324MHz
PHY drv:clk:38,ca:38,DQ:30,odt:0
vrefinner:41%, vrefout:41%
dram drv:40,odt:0
clk skew:0x61
rx vref: 43.9%
tx vref: 38.0%

change to: 528MHz
PHY drv:clk:38,ca:38,DQ:30,odt:0
vrefinner:41%, vrefout:41%
dram drv:40,odt:0
clk skew:0x58
rx vref: 39.9%
tx vref: 38.0%

change to: 780MHz
PHY drv:clk:38,ca:38,DQ:30,odt:60
vrefinner:16%, vrefout:41%
dram drv:40,odt:0
clk skew:0x58
rx vref: 16.4%
tx vref: 38.0%

change to: 1560MHz(final freq)
PHY drv:clk:38,ca:38,DQ:30,odt:60
vrefinner:16%, vrefout:29%
dram drv:40,odt:80
vref_ca:00000068
clk skew:0x1c
rx vref: 17.4%
tx vref: 31.0%
cs 0:
the read training result:
DQS0:0x34, DQS1:0x35, DQS2:0x35, DQS3:0x32, 
min  : 0xf 0x13 0x14  0xf  0x1  0x6  0xa  0x5 , 0xb  0xa  0x2  0x1  0xd  0xc  0xb  0xa ,
      0x10 0x12  0xc  0xd  0x4  0x1  0x2  0x6 , 0xe  0xa  0xa  0x3 0x10 0x15  0xd 0x13 ,
mid  :0x2a 0x2b 0x2d 0x2a 0x1c 0x21 0x24 0x20 ,0x27 0x25 0x1b 0x1d 0x28 0x25 0x25 0x25 ,
      0x2c 0x2d 0x25 0x25 0x1e 0x1a 0x1b 0x20 ,0x28 0x24 0x24 0x1d 0x2a 0x2e 0x28 0x2c ,
max  :0x45 0x44 0x46 0x45 0x37 0x3d 0x3f 0x3c ,0x44 0x40 0x35 0x39 0x44 0x3f 0x3f 0x41 ,
      0x49 0x48 0x3f 0x3d 0x38 0x34 0x34 0x3a ,0x42 0x3e 0x3f 0x37 0x45 0x48 0x43 0x46 ,
range:0x36 0x31 0x32 0x36 0x36 0x37 0x35 0x37 ,0x39 0x36 0x33 0x38 0x37 0x33 0x34 0x37 ,
      0x39 0x36 0x33 0x30 0x34 0x33 0x32 0x34 ,0x34 0x34 0x35 0x34 0x35 0x33 0x36 0x33 ,
the write training result:
DQS0:0x25, DQS1:0xe, DQS2:0x18, DQS3:0x5, 
min  :0x79 0x7c 0x7d 0x79 0x6b 0x6f 0x73 0x70 0x6f ,0x5e 0x5c 0x53 0x54 0x60 0x5e 0x5c 0x5b 0x59 ,
      0x69 0x6a 0x63 0x62 0x5c 0x5a 0x59 0x5e 0x60 ,0x59 0x57 0x54 0x4f 0x5c 0x5d 0x57 0x5d 0x53 ,
mid  :0x93 0x95 0x97 0x93 0x84 0x88 0x8d 0x89 0x89 ,0x77 0x75 0x6d 0x6d 0x79 0x76 0x76 0x75 0x72 ,
      0x85 0x86 0x7c 0x7b 0x76 0x73 0x72 0x76 0x79 ,0x72 0x70 0x6d 0x68 0x75 0x77 0x71 0x76 0x6d ,
max  :0xae 0xae 0xb2 0xae 0x9d 0xa2 0xa7 0xa3 0xa3 ,0x90 0x8f 0x87 0x87 0x93 0x8f 0x91 0x8f 0x8c ,
      0xa1 0xa2 0x95 0x94 0x90 0x8c 0x8c 0x8f 0x93 ,0x8c 0x8a 0x87 0x82 0x8f 0x91 0x8c 0x90 0x88 ,
range:0x35 0x32 0x35 0x35 0x32 0x33 0x34 0x33 0x34 ,0x32 0x33 0x34 0x33 0x33 0x31 0x35 0x34 0x33 ,
      0x38 0x38 0x32 0x32 0x34 0x32 0x33 0x31 0x33 ,0x33 0x33 0x33 0x33 0x33 0x34 0x35 0x33 0x35 ,
cs 1:
the read training result:
DQS0:0x34, DQS1:0x35, DQS2:0x35, DQS3:0x32, 
min  : 0xf 0x13 0x14  0xf  0x1  0x6  0xa  0x5 , 0xb  0xa  0x2  0x1  0xd  0xc  0xb  0xa ,
      0x10 0x12  0xc  0xd  0x4  0x1  0x2  0x6 , 0xe  0xa  0xa  0x3 0x10 0x15  0xd 0x13 ,
mid  :0x2a 0x2b 0x2d 0x2a 0x1c 0x21 0x24 0x20 ,0x27 0x25 0x1b 0x1d 0x28 0x25 0x25 0x25 ,
      0x2c 0x2d 0x25 0x25 0x1e 0x1a 0x1b 0x20 ,0x28 0x24 0x24 0x1d 0x2a 0x2e 0x28 0x2c ,
max  :0x45 0x44 0x46 0x45 0x37 0x3d 0x3f 0x3c ,0x44 0x40 0x35 0x39 0x44 0x3f 0x3f 0x41 ,
      0x49 0x48 0x3f 0x3d 0x38 0x34 0x34 0x3a ,0x42 0x3e 0x3f 0x37 0x45 0x48 0x43 0x46 ,
range:0x36 0x31 0x32 0x36 0x36 0x37 0x35 0x37 ,0x39 0x36 0x33 0x38 0x37 0x33 0x34 0x37 ,
      0x39 0x36 0x33 0x30 0x34 0x33 0x32 0x34 ,0x34 0x34 0x35 0x34 0x35 0x33 0x36 0x33 ,
the write training result:
DQS0:0x25, DQS1:0xe, DQS2:0x18, DQS3:0x5, 
min  :0x79 0x7c 0x7d 0x79 0x6b 0x6f 0x73 0x70 0x6f ,0x5e 0x5c 0x53 0x54 0x60 0x5e 0x5c 0x5b 0x59 ,
      0x69 0x6a 0x63 0x62 0x5c 0x5a 0x59 0x5e 0x60 ,0x59 0x57 0x54 0x4f 0x5c 0x5d 0x57 0x5d 0x53 ,
mid  :0x93 0x95 0x97 0x93 0x84 0x88 0x8d 0x89 0x89 ,0x77 0x75 0x6d 0x6d 0x79 0x76 0x76 0x75 0x72 ,
      0x85 0x86 0x7c 0x7b 0x76 0x73 0x72 0x76 0x79 ,0x72 0x70 0x6d 0x68 0x75 0x77 0x71 0x76 0x6d ,
max  :0xae 0xae 0xb2 0xae 0x9d 0xa2 0xa7 0xa3 0xa3 ,0x90 0x8f 0x87 0x87 0x93 0x8f 0x91 0x8f 0x8c ,
      0xa1 0xa2 0x95 0x94 0x90 0x8c 0x8c 0x8f 0x93 ,0x8c 0x8a 0x87 0x82 0x8f 0x91 0x8c 0x90 0x88 ,
range:0x35 0x32 0x35 0x35 0x32 0x33 0x34 0x33 0x34 ,0x32 0x33 0x34 0x33 0x33 0x31 0x35 0x34 0x33 ,
      0x38 0x38 0x32 0x32 0x34 0x32 0x33 0x31 0x33 ,0x33 0x33 0x33 0x33 0x33 0x34 0x35 0x33 0x35 ,
CA Training result:
cs:0 min  :0x51 0x4d 0x45 0x43 0x49 0x40 0x4d ,0x52 0x4d 0x48 0x40 0x44 0x40 0x4d ,
cs:0 mid  :0x8e 0x8d 0x82 0x82 0x85 0x7f 0x78 ,0x8f 0x8c 0x84 0x7f 0x80 0x7f 0x78 ,
cs:0 max  :0xcc 0xce 0xbf 0xc2 0xc1 0xbf 0xa4 ,0xcc 0xcc 0xc0 0xbf 0xbd 0xbf 0xa4 ,
cs:0 range:0x7b 0x81 0x7a 0x7f 0x78 0x7f 0x57 ,0x7a 0x7f 0x78 0x7f 0x79 0x7f 0x57 ,
cs:1 min  :0x53 0x50 0x49 0x43 0x4a 0x3f 0x4c ,0x56 0x51 0x4b 0x45 0x48 0x45 0x4e ,
cs:1 mid  :0x91 0x8f 0x86 0x82 0x86 0x7e 0x7c ,0x93 0x90 0x87 0x83 0x84 0x83 0x7c ,
cs:1 max  :0xcf 0xcf 0xc4 0xc2 0xc3 0xbe 0xac ,0xd1 0xcf 0xc4 0xc1 0xc0 0xc2 0xab ,
cs:1 range:0x7c 0x7f 0x7b 0x7f 0x79 0x7f 0x60 ,0x7b 0x7e 0x79 0x7c 0x78 0x7d 0x5d ,
out

U-Boot SPL 2024.10-OpenWrt-r0-300ebaf (Nov 12 2025 - 08:28:48 +0000)
Trying to boot from MMC1
## Checking hash(es) for config config-1 ... OK
## Checking hash(es) for Image atf-1 ... sha256+ OK
## Checking hash(es) for Image u-boot ... sha256+ OK
## Checking hash(es) for Image fdt-1 ... sha256+ OK
## Checking hash(es) for Image atf-2 ... sha256+ OK
## Checking hash(es) for Image atf-3 ... sha256+ OK
## Checking hash(es) for Image atf-4 ... sha256+ OK
## Checking hash(es) for Image atf-5 ... sha256+ OK
## Checking hash(es) for Image atf-6 ... sha256+ OK
INFO:    Preloader serial: 2
NOTICE:  BL31: v2.3():v2.3-645-g8cea6ab0b:cl, fwver: v1.44
NOTICE:  BL31: Built : 16:36:43, Sep 19 2023
INFO:    GICv3 without legacy support detected.
INFO:    ARM GICv3 driver initialized in EL3
INFO:    pmu v1 is valid 220114
INFO:    l3 cache partition cfg-0
INFO:    dfs DDR fsp_param[0].freq_mhz= 1560MHz
INFO:    dfs DDR fsp_param[1].freq_mhz= 324MHz
INFO:    dfs DDR fsp_param[2].freq_mhz= 528MHz
INFO:    dfs DDR fsp_param[3].freq_mhz= 780MHz
INFO:    Using opteed sec cpu_context!
INFO:    boot cpu mask: 0
INFO:    BL31: Initializing runtime services
WARNING: No OPTEE provided by BL2 boot loader, Booting device without OPTEE initialization. SMC`s destined for OPTEE will return SMC_UNK
ERROR:   Error initializing runtime service opteed_fast
INFO:    BL31: Preparing for EL3 exit to normal world
INFO:    Entry point address = 0xa00000
INFO:    SPSR = 0x3c9


U-Boot 2024.10-OpenWrt-r0-300ebaf (Nov 12 2025 - 08:28:48 +0000)

Model: Generic RK3566/RK3568
DRAM:  4 GiB
Core:  250 devices, 25 uclasses, devicetree: separate
MMC:   mmc@fe2b0000: 1, mmc@fe310000: 0
Loading Environment from nowhere... OK
In:    serial@fe660000
Out:   serial@fe660000
Err:   serial@fe660000
Model: Generic RK3566/RK3568
Hit any key to stop autoboot:  0 
Scanning for bootflows in all bootdevs
Seq  Method       State   Uclass    Part  Name                      Filename
---  -----------  ------  --------  ----  ------------------------  ----------------
Scanning global bootmeth 'efi_mgr':
Card did not respond to voltage select! : -110
No EFI system partition
No EFI system partition
Failed to persist EFI variables
No EFI system partition
Failed to persist EFI variables
No EFI system partition
Failed to persist EFI variables
  0  efi_mgr      ready   (none)       0  <NULL>                    
** Booting bootflow '<NULL>' with efi_mgr
Loading Boot0000 'mmc 0' failed
EFI boot manager: Cannot load any image
Boot failed (err=-14)
Scanning bootdev 'mmc@fe2b0000.bootdev':
Card did not respond to voltage select! : -110
Scanning bootdev 'mmc@fe310000.bootdev':
  1  script       ready   mmc          1  mmc@fe310000.bootdev.part /boot.scr
** Booting bootflow 'mmc@fe310000.bootdev.part_1' with script
4756656 bytes read in 37 ms (122.6 MiB/s)
## Loading kernel from FIT Image at 02000000 ...
   Using 'config-1' configuration
   Verifying Hash Integrity ... OK
   Trying 'kernel-1' kernel subimage
     Description:  ARM64 OpenWrt Linux-6.6.116
     Type:         Kernel Image
     Compression:  lzma compressed
     Data Start:   0x020000e8
     Data Size:    4695021 Bytes = 4.5 MiB
     Architecture: AArch64
     OS:           Linux
     Load Address: 0x03200000
     Entry Point:  0x03200000
     Hash algo:    crc32
     Hash value:   0472ea40
     Hash algo:    sha1
     Hash value:   8dfb9cc787682e554dac9dd554efa2de0fe3ec96
   Verifying Hash Integrity ... crc32+ sha1+ OK
## Loading fdt from FIT Image at 02000000 ...
   Using 'config-1' configuration
   Verifying Hash Integrity ... OK
   Trying 'fdt-1' fdt subimage
     Description:  ARM64 OpenWrt linkease_easepi-r1 device tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x0247a61c
     Data Size:    59724 Bytes = 58.3 KiB
     Architecture: AArch64
     Hash algo:    crc32
     Hash value:   918bdad4
     Hash algo:    sha1
     Hash value:   e05310a2aa3d152d33aa790e83b5cc9ae99abb6c
   Verifying Hash Integrity ... crc32+ sha1+ OK
   Booting using the fdt blob at 0x247a61c
Working FDT set to 247a61c
   Uncompressing Kernel Image to 3200000
   Loading Device Tree to 00000000ececc000, end 00000000ecedd94b ... OK
Working FDT set to ececc000

Starting kernel ...

[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x412fd050]
[    0.000000] Linux version 6.6.116 (build@buildserver) (aarch64-openwrt-linux-musl-gcc (OpenWrt GCC 13.3.0 r0-300ebaf) 13.3.0, GNU ld (GNU Binutils) 2.42) #0 SMP Wed Nov 12 08:28:48 2025
[    0.000000] Machine model: LinkEase EasePi R1
[    0.000000] earlycon: uart8250 at MMIO32 0x00000000fe660000 (options '')
[    0.000000] printk: bootconsole [uart8250] enabled
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000200000-0x00000000ffffffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   [mem 0x0000000100000000-0x00000001ffffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000200000-0x00000000efffffff]
[    0.000000]   node   0: [mem 0x00000001f0000000-0x00000001ffffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000200000-0x00000001ffffffff]
[    0.000000] On node 0, zone DMA: 512 pages in unavailable ranges
[    0.000000] cma: Reserved 16 MiB at 0x00000000ef000000 on node -1
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.1 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.2
[    0.000000] percpu: Embedded 19 pages/cpu s37736 r8192 d31896 u77824
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: GIC system register CPU interface
[    0.000000] CPU features: detected: Virtualization Host Extensions
[    0.000000] CPU features: detected: Qualcomm erratum 1009, or ARM erratum 1286807, 2441009
[    0.000000] CPU features: detected: ARM errata 1165522, 1319367, or 1530923
[    0.000000] alternatives: applying boot alternatives
[    0.000000] Kernel command line: console=ttyS2,1500000 earlycon=uart8250,mmio32,0xfe660000 root=PARTUUID=5452574f-02 rw rootwait
[    0.000000] Dentry cache hash table entries: 524288 (order: 10, 4194304 bytes, linear)
[    0.000000] Inode-cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 1031688
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] software IO TLB: area num 4.
[    0.000000] software IO TLB: mapped [mem 0x00000000e8ecc000-0x00000000ececc000] (64MB)
[    0.000000] Memory: 4010872K/4192256K available (9600K kernel code, 2444K rwdata, 2884K rodata, 2240K init, 731K bss, 165000K reserved, 16384K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu: 	RCU event tracing is enabled.
[    0.000000] rcu: 	RCU restricting CPUs from NR_CPUS=256 to nr_cpu_ids=4.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] GIC: enabling workaround for GICv3: non-coherent attribute
[    0.000000] GICv3: GIC: Using split EOI/Deactivate mode
[    0.000000] GICv3: 320 SPIs implemented
[    0.000000] GICv3: 0 Extended SPIs implemented
[    0.000000] GICv3: MBI range [296:319]
[    0.000000] GICv3: Using MBI frame 0x00000000fd410000
[    0.000000] Root IRQ handler: gic_handle_irq
[    0.000000] GICv3: GICv3 features: 16 PPIs
[    0.000000] GICv3: CPU0: found redistributor 0 region 0:0x00000000fd460000
[    0.000000] ITS [mem 0xfd440000-0xfd45ffff]
[    0.000000] GIC: enabling workaround for ITS: Rockchip erratum RK3568002
[    0.000000] GIC: enabling workaround for ITS: non-coherent attribute
[    0.000000] ITS@0x00000000fd440000: allocated 8192 Devices @410000 (indirect, esz 8, psz 64K, shr 0)
[    0.000000] ITS@0x00000000fd440000: allocated 32768 Interrupt Collections @420000 (flat, esz 2, psz 64K, shr 0)
[    0.000000] ITS: using cache flushing for cmd queue
[    0.000000] GICv3: using LPI property table @0x0000000000430000
[    0.000000] GIC: using cache flushing for LPI property table
[    0.000000] GICv3: CPU0: using allocated LPI pending table @0x0000000000440000
[    0.000000] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.000000] arch_timer: cp15 timer(s) running at 24.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000000] sched_clock: 56 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.001461] Console: colour dummy device 80x25
[    0.001962] Calibrating delay loop (skipped), value calculated using timer frequency.. 48.00 BogoMIPS (lpj=240000)
[    0.002981] pid_max: default: 32768 minimum: 301
[    0.008346] Mount-cache hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.009098] Mountpoint-cache hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.014592] RCU Tasks Trace: Setting shift to 2 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=4.
[    0.015774] rcu: Hierarchical SRCU implementation.
[    0.016249] rcu: 	Max phase no-delay instances is 1000.
[    0.017142] Platform MSI: msi-controller@fd440000 domain created
[    0.018053] PCI/MSI: /interrupt-controller@fd400000/msi-controller@fd440000 domain created
[    0.019541] smp: Bringing up secondary CPUs ...
[    0.020746] Detected VIPT I-cache on CPU1
[    0.020833] GICv3: CPU1: found redistributor 100 region 0:0x00000000fd480000
[    0.020855] GICv3: CPU1: using allocated LPI pending table @0x0000000000450000
[    0.020911] CPU1: Booted secondary processor 0x0000000100 [0x412fd050]
[    0.021730] Detected VIPT I-cache on CPU2
[    0.021809] GICv3: CPU2: found redistributor 200 region 0:0x00000000fd4a0000
[    0.021829] GICv3: CPU2: using allocated LPI pending table @0x0000000000460000
[    0.021872] CPU2: Booted secondary processor 0x0000000200 [0x412fd050]
[    0.022703] Detected VIPT I-cache on CPU3
[    0.022777] GICv3: CPU3: found redistributor 300 region 0:0x00000000fd4c0000
[    0.022797] GICv3: CPU3: using allocated LPI pending table @0x0000000000470000
[    0.022837] CPU3: Booted secondary processor 0x0000000300 [0x412fd050]
[    0.022964] smp: Brought up 1 node, 4 CPUs
[    0.030637] SMP: Total of 4 processors activated.
[    0.031101] CPU features: detected: 32-bit EL0 Support
[    0.031606] CPU features: detected: Data cache clean to the PoU not required for I/D coherence
[    0.032446] CPU features: detected: Common not Private translations
[    0.033058] CPU features: detected: CRC32 instructions
[    0.033564] CPU features: detected: RCpc load-acquire (LDAPR)
[    0.034126] CPU features: detected: Privileged Access Never
[    0.034670] CPU features: detected: RAS Extension Support
[    0.035200] CPU features: detected: Speculative Store Bypassing Safe (SSBS)
[    0.035988] CPU: All CPU(s) started at EL2
[    0.036396] alternatives: applying system-wide alternatives
[    0.051992] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.052982] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.056075] pinctrl core: initialized pinctrl subsystem
[    0.059199] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.060839] DMA: preallocated 512 KiB GFP_KERNEL pool for atomic allocations
[    0.061720] DMA: preallocated 512 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.062729] DMA: preallocated 512 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.065596] thermal_sys: Registered thermal governor 'step_wise'
[    0.065607] thermal_sys: Registered thermal governor 'power_allocator'
[    0.066255] cpuidle: using governor menu
[    0.067417] ASID allocator initialised with 65536 entries
[    0.068170] Serial: AMBA PL011 UART driver
[    0.080050] /vop@fe040000: Fixed dependency cycle(s) with /hdmi@fe0a0000
[    0.080820] /hdmi@fe0a0000: Fixed dependency cycle(s) with /vop@fe040000
[    0.093125] gpio gpiochip0: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.094268] rockchip-gpio fdd60000.gpio: probed /pinctrl/gpio@fdd60000
[    0.095292] gpio gpiochip1: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.096388] rockchip-gpio fe740000.gpio: probed /pinctrl/gpio@fe740000
[    0.097479] gpio gpiochip2: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.098553] rockchip-gpio fe750000.gpio: probed /pinctrl/gpio@fe750000
[    0.099700] gpio gpiochip3: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.100773] rockchip-gpio fe760000.gpio: probed /pinctrl/gpio@fe760000
[    0.101763] gpio gpiochip4: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.102842] rockchip-gpio fe770000.gpio: probed /pinctrl/gpio@fe770000
[    0.107956] /hdmi@fe0a0000: Fixed dependency cycle(s) with /hdmi-con
[    0.108666] /hdmi-con: Fixed dependency cycle(s) with /hdmi@fe0a0000
[    0.112313] Modules: 28256 pages in range for non-PLT usage
[    0.112325] Modules: 519776 pages in range for PLT usage
[    0.114282] HugeTLB: registered 1.00 GiB page size, pre-allocated 0 pages
[    0.115469] HugeTLB: 0 KiB vmemmap can be freed for a 1.00 GiB page
[    0.116085] HugeTLB: registered 32.0 MiB page size, pre-allocated 0 pages
[    0.116747] HugeTLB: 0 KiB vmemmap can be freed for a 32.0 MiB page
[    0.117361] HugeTLB: registered 2.00 MiB page size, pre-allocated 0 pages
[    0.118024] HugeTLB: 0 KiB vmemmap can be freed for a 2.00 MiB page
[    0.118637] HugeTLB: registered 64.0 KiB page size, pre-allocated 0 pages
[    0.119298] HugeTLB: 0 KiB vmemmap can be freed for a 64.0 KiB page
[    0.122408] cryptd: max_cpu_qlen set to 1000
[    0.126374] iommu: Default domain type: Passthrough
[    0.127283] SCSI subsystem initialized
[    0.128205] usbcore: registered new interface driver usbfs
[    0.128801] usbcore: registered new interface driver hub
[    0.129378] usbcore: registered new device driver usb
[    0.130343] pps_core: LinuxPPS API ver. 1 registered
[    0.130834] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.131740] PTP clock support registered
[    0.132620] scmi_core: SCMI protocol bus registered
[    0.134807] clocksource: Switched to clocksource arch_sys_counter
[    0.137666] VFS: Disk quotas dquot_6.6.0
[    0.138236] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.145562] NET: Registered PF_INET protocol family
[    0.146355] IP idents hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.151955] tcp_listen_portaddr_hash hash table entries: 2048 (order: 3, 32768 bytes, linear)
[    0.152833] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.153609] TCP established hash table entries: 32768 (order: 6, 262144 bytes, linear)
[    0.154647] TCP bind hash table entries: 32768 (order: 8, 1048576 bytes, linear)
[    0.156325] TCP: Hash tables configured (established 32768 bind 32768)
[    0.157571] MPTCP token hash table entries: 4096 (order: 4, 98304 bytes, linear)
[    0.158556] UDP hash table entries: 2048 (order: 4, 65536 bytes, linear)
[    0.159294] UDP-Lite hash table entries: 2048 (order: 4, 65536 bytes, linear)
[    0.160404] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.161002] PCI: CLS 0 bytes, default 64
[    0.163428] workingset: timestamp_bits=46 max_order=20 bucket_order=0
[    0.172134] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.172715] jffs2: version 2.2 (NAND) (SUMMARY) (ZLIB) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.178973] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 248)
[    0.179894] io scheduler mq-deadline registered
[    0.188828] rockchip-dw-pcie 3c0000000.pcie: host bridge /pcie@fe260000 ranges:
[    0.189588] rockchip-dw-pcie 3c0000000.pcie:       IO 0x00f4100000..0x00f41fffff -> 0x00f4100000
[    0.190468] rockchip-dw-pcie 3c0000000.pcie:      MEM 0x00f4200000..0x00f5ffffff -> 0x00f4200000
[    0.191337] rockchip-dw-pcie 3c0000000.pcie:      MEM 0x0300000000..0x033fffffff -> 0x0040000000
[    0.192408] rockchip-dw-pcie 3c0000000.pcie: iATU: unroll T, 8 ob, 8 ib, align 64K, limit 8G
[    0.404857] rockchip-dw-pcie 3c0000000.pcie: PCIe Gen.2 x1 link up
[    0.405800] rockchip-dw-pcie 3c0000000.pcie: PCI host bridge to bus 0000:00
[    0.406490] pci_bus 0000:00: root bus resource [bus 00-0f]
[    0.407037] pci_bus 0000:00: root bus resource [io  0x0000-0xfffff] (bus address [0xf4100000-0xf41fffff])
[    0.407974] pci_bus 0000:00: root bus resource [mem 0xf4200000-0xf5ffffff]
[    0.408653] pci_bus 0000:00: root bus resource [mem 0x300000000-0x33fffffff] (bus address [0x40000000-0x7fffffff])
[    0.409700] pci 0000:00:00.0: [1d87:3566] type 01 class 0x060400
[    0.410316] pci 0000:00:00.0: reg 0x38: [mem 0x00000000-0x0000ffff pref]
[    0.411043] pci 0000:00:00.0: supports D1 D2
[    0.411468] pci 0000:00:00.0: PME# supported from D0 D1 D3hot
[    0.417101] pci_bus 0000:01: busn_res: can not insert [bus 01-ff] under [bus 00-0f] (conflicts with (null) [bus 00-0f])
[    0.418275] pci 0000:01:00.0: [10ec:8125] type 00 class 0x020000
[    0.418943] pci 0000:01:00.0: reg 0x10: initial BAR value 0x00000000 invalid
[    0.419638] pci 0000:01:00.0: reg 0x10: [io  size 0x0100]
[    0.420255] pci 0000:01:00.0: reg 0x18: [mem 0x00000000-0x0000ffff 64bit]
[    0.420977] pci 0000:01:00.0: reg 0x20: [mem 0x00000000-0x00003fff 64bit]
[    0.422144] pci 0000:01:00.0: supports D1 D2
[    0.422568] pci 0000:01:00.0: PME# supported from D0 D1 D2 D3hot D3cold
[    0.444986] pci 0000:00:00.0: BAR 8: assigned [mem 0xf4200000-0xf42fffff]
[    0.445666] pci 0000:00:00.0: BAR 6: assigned [mem 0xf4300000-0xf430ffff pref]
[    0.446381] pci 0000:00:00.0: BAR 7: assigned [io  0x1000-0x1fff]
[    0.446989] pci 0000:01:00.0: BAR 2: assigned [mem 0xf4200000-0xf420ffff 64bit]
[    0.447755] pci 0000:01:00.0: BAR 4: assigned [mem 0xf4210000-0xf4213fff 64bit]
[    0.448518] pci 0000:01:00.0: BAR 0: assigned [io  0x1000-0x10ff]
[    0.449133] pci 0000:00:00.0: PCI bridge to [bus 01-ff]
[    0.449651] pci 0000:00:00.0:   bridge window [io  0x1000-0x1fff]
[    0.450253] pci 0000:00:00.0:   bridge window [mem 0xf4200000-0xf42fffff]
[    0.454880] pcieport 0000:00:00.0: PME: Signaling with IRQ 22
[    0.455856] pcieport 0000:00:00.0: AER: enabled with IRQ 31
[    0.457484] phy phy-fe8c0000.phy.2: lane number 0, val 1
[    0.458015] phy phy-fe8c0000.phy.2: lane number 1, val 2
[    0.458540] phy phy-fe8c0000.phy.2: bifurcation enabled
[    0.459187] rockchip-dw-pcie 3c0400000.pcie: host bridge /pcie@fe270000 ranges:
[    0.459936] rockchip-dw-pcie 3c0400000.pcie:       IO 0x00f2100000..0x00f21fffff -> 0x00f2100000
[    0.460833] rockchip-dw-pcie 3c0400000.pcie:      MEM 0x00f2200000..0x00f3ffffff -> 0x00f2200000
[    0.461704] rockchip-dw-pcie 3c0400000.pcie:      MEM 0x0340000000..0x037fffffff -> 0x0040000000
[    0.471035] rockchip-dw-pcie 3c0400000.pcie: iATU: unroll T, 8 ob, 8 ib, align 64K, limit 8G
[    0.684857] rockchip-dw-pcie 3c0400000.pcie: PCIe Gen.2 x1 link up
[    0.685639] rockchip-dw-pcie 3c0400000.pcie: PCI host bridge to bus 0001:10
[    0.686331] pci_bus 0001:10: root bus resource [bus 10-1f]
[    0.686878] pci_bus 0001:10: root bus resource [io  0x100000-0x1fffff] (bus address [0xf2100000-0xf21fffff])
[    0.687840] pci_bus 0001:10: root bus resource [mem 0xf2200000-0xf3ffffff]
[    0.688517] pci_bus 0001:10: root bus resource [mem 0x340000000-0x37fffffff] (bus address [0x40000000-0x7fffffff])
[    0.689563] pci 0001:10:00.0: [1d87:3566] type 01 class 0x060400
[    0.690179] pci 0001:10:00.0: reg 0x38: [mem 0x00000000-0x0000ffff pref]
[    0.690906] pci 0001:10:00.0: supports D1 D2
[    0.691331] pci 0001:10:00.0: PME# supported from D0 D1 D3hot
[    0.696908] pci 0001:10:00.0: Primary bus is hard wired to 0
[    0.697475] pci 0001:10:00.0: bridge configuration invalid ([bus 01-ff]), reconfiguring
[    0.698480] pci 0001:11:00.0: [10ec:8125] type 00 class 0x020000
[    0.699149] pci 0001:11:00.0: reg 0x10: [io  0x0000-0x00ff]
[    0.699785] pci 0001:11:00.0: reg 0x18: [mem 0x00000000-0x0000ffff 64bit]
[    0.700507] pci 0001:11:00.0: reg 0x20: [mem 0x00000000-0x00003fff 64bit]
[    0.701671] pci 0001:11:00.0: supports D1 D2
[    0.702096] pci 0001:11:00.0: PME# supported from D0 D1 D2 D3hot D3cold
[    0.724958] pci_bus 0001:11: busn_res: [bus 11-1f] end is updated to 11
[    0.725643] pci 0001:10:00.0: BAR 8: assigned [mem 0xf2200000-0xf22fffff]
[    0.726316] pci 0001:10:00.0: BAR 6: assigned [mem 0xf2300000-0xf230ffff pref]
[    0.727030] pci 0001:10:00.0: BAR 7: assigned [io  0x100000-0x100fff]
[    0.727671] pci 0001:11:00.0: BAR 2: assigned [mem 0xf2200000-0xf220ffff 64bit]
[    0.728435] pci 0001:11:00.0: BAR 4: assigned [mem 0xf2210000-0xf2213fff 64bit]
[    0.729197] pci 0001:11:00.0: BAR 0: assigned [io  0x100000-0x1000ff]
[    0.729845] pci 0001:10:00.0: PCI bridge to [bus 11]
[    0.730336] pci 0001:10:00.0:   bridge window [io  0x100000-0x100fff]
[    0.730972] pci 0001:10:00.0:   bridge window [mem 0xf2200000-0xf22fffff]
[    0.735282] pcieport 0001:10:00.0: PME: Signaling with IRQ 42
[    0.736303] pcieport 0001:10:00.0: AER: enabled with IRQ 43
[    0.737951] rockchip-dw-pcie 3c0800000.pcie: host bridge /pcie@fe280000 ranges:
[    0.738703] rockchip-dw-pcie 3c0800000.pcie:       IO 0x00f0100000..0x00f01fffff -> 0x00f0100000
[    0.739601] rockchip-dw-pcie 3c0800000.pcie:      MEM 0x00f0200000..0x00f1ffffff -> 0x00f0200000
[    0.740473] rockchip-dw-pcie 3c0800000.pcie:      MEM 0x0380000000..0x03bfffffff -> 0x0040000000
[    0.749645] rockchip-dw-pcie 3c0800000.pcie: iATU: unroll T, 8 ob, 8 ib, align 64K, limit 8G
[    0.964854] rockchip-dw-pcie 3c0800000.pcie: PCIe Gen.3 x1 link up
[    0.965611] rockchip-dw-pcie 3c0800000.pcie: PCI host bridge to bus 0002:20
[    0.966302] pci_bus 0002:20: root bus resource [bus 20-2f]
[    0.966849] pci_bus 0002:20: root bus resource [io  0x200000-0x2fffff] (bus address [0xf0100000-0xf01fffff])
[    0.967812] pci_bus 0002:20: root bus resource [mem 0xf0200000-0xf1ffffff]
[    0.968489] pci_bus 0002:20: root bus resource [mem 0x380000000-0x3bfffffff] (bus address [0x40000000-0x7fffffff])
[    0.969533] pci 0002:20:00.0: [1d87:3566] type 01 class 0x060400
[    0.970139] pci 0002:20:00.0: reg 0x10: [mem 0x00000000-0x3fffffff]
[    0.970761] pci 0002:20:00.0: reg 0x14: [mem 0x00000000-0x3fffffff]
[    0.971383] pci 0002:20:00.0: reg 0x38: [mem 0x00000000-0x0000ffff pref]
[    0.972105] pci 0002:20:00.0: supports D1 D2
[    0.972529] pci 0002:20:00.0: PME# supported from D0 D1 D3hot
[    0.977892] pci 0002:20:00.0: Primary bus is hard wired to 0
[    0.978458] pci 0002:20:00.0: bridge configuration invalid ([bus 01-ff]), reconfiguring
[    0.979451] pci 0002:21:00.0: [126f:2263] type 00 class 0x010802
[    0.980138] pci 0002:21:00.0: reg 0x10: [mem 0x00000000-0x00003fff 64bit]
[    0.981434] pci 0002:21:00.0: 7.876 Gb/s available PCIe bandwidth, limited by 8.0 GT/s PCIe x1 link at 0002:20:00.0 (capable of 31.504 Gb/s with 8.0 GT/s PCIe x4 link)
[    1.004954] pci_bus 0002:21: busn_res: [bus 21-2f] end is updated to 21
[    1.005640] pci 0002:20:00.0: BAR 0: assigned [mem 0x380000000-0x3bfffffff]
[    1.006333] pci 0002:20:00.0: BAR 1: no space for [mem size 0x40000000]
[    1.006985] pci 0002:20:00.0: BAR 1: failed to assign [mem size 0x40000000]
[    1.007672] pci 0002:20:00.0: BAR 8: assigned [mem 0xf0200000-0xf02fffff]
[    1.008340] pci 0002:20:00.0: BAR 6: assigned [mem 0xf0300000-0xf030ffff pref]
[    1.009055] pci 0002:21:00.0: BAR 0: assigned [mem 0xf0200000-0xf0203fff 64bit]
[    1.009815] pci 0002:20:00.0: PCI bridge to [bus 21]
[    1.010309] pci 0002:20:00.0:   bridge window [mem 0xf0200000-0xf02fffff]
[    1.014562] pcieport 0002:20:00.0: PME: Signaling with IRQ 54
[    1.015634] pcieport 0002:20:00.0: AER: enabled with IRQ 55
[    1.020720] dma-pl330 fe530000.dma-controller: Loaded driver for PL330 DMAC-241330
[    1.021475] dma-pl330 fe530000.dma-controller: 	DBUFF-128x8bytes Num_Chans-8 Num_Peri-32 Num_Events-16
[    1.025069] dma-pl330 fe550000.dma-controller: Loaded driver for PL330 DMAC-241330
[    1.025819] dma-pl330 fe550000.dma-controller: 	DBUFF-128x8bytes Num_Chans-8 Num_Peri-32 Num_Events-16
[    1.028260] Serial: 8250/16550 driver, 10 ports, IRQ sharing enabled
[    1.032968] printk: console [ttyS2] disabled
[    1.033763] fe660000.serial: ttyS2 at MMIO 0xfe660000 (irq = 60, base_baud = 1500000) is a 16550A
[    1.034961] printk: console [ttyS2] enabled
[    1.034961] printk: console [ttyS2] enabled
[    1.035749] printk: bootconsole [uart8250] disabled
[    1.035749] printk: bootconsole [uart8250] disabled
[    1.042242] random: crng init done
[    1.047740] loop: module loaded
[    1.048042] mtip32xx Version 1.3.1
[    1.049874] nvme nvme0: pci function 0002:21:00.0
[    1.050328] nvme 0002:21:00.0: enabling device (0000 -> 0002)
[    1.054097] rk_gmac-dwmac fe010000.ethernet: IRQ eth_lpi not found
[    1.054924] rk_gmac-dwmac fe010000.ethernet: supply phy not found, using dummy regulator
[    1.055779] rk_gmac-dwmac fe010000.ethernet: Can not read property: clock_in_out.
[    1.056448] rk_gmac-dwmac fe010000.ethernet: Can not read property: tx_delay.
[    1.057079] rk_gmac-dwmac fe010000.ethernet: set tx_delay to 0x30
[    1.057622] rk_gmac-dwmac fe010000.ethernet: Can not read property: rx_delay.
[    1.058250] rk_gmac-dwmac fe010000.ethernet: set rx_delay to 0x10
[    1.058801] rk_gmac-dwmac fe010000.ethernet: integrated PHY? (no).
[    1.059388] rk_gmac-dwmac fe010000.ethernet: clock input from PHY
[    1.061157] hwmon hwmon0: temp1_input not attached to any thermal zone
[    1.064945] rk_gmac-dwmac fe010000.ethernet: init for RGMII_ID
[    1.065788] rk_gmac-dwmac fe010000.ethernet: User ID: 0x30, Synopsys ID: 0x51
[    1.066440] rk_gmac-dwmac fe010000.ethernet: 	DWMAC4/5
[    1.066903] rk_gmac-dwmac fe010000.ethernet: DMA HW capability register supported
[    1.067564] rk_gmac-dwmac fe010000.ethernet: RX Checksum Offload Engine supported
[    1.068222] rk_gmac-dwmac fe010000.ethernet: TX Checksum insertion supported
[    1.068842] rk_gmac-dwmac fe010000.ethernet: Wake-Up On Lan supported
[    1.069411] rk_gmac-dwmac fe010000.ethernet: TSO supported
[    1.069897] rk_gmac-dwmac fe010000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    1.070603] rk_gmac-dwmac fe010000.ethernet: Enabled RFS Flow TC (entries=10)
[    1.071235] rk_gmac-dwmac fe010000.ethernet: TSO feature enabled
[    1.071769] rk_gmac-dwmac fe010000.ethernet: Using 32/32 bits DMA host/device width
[    1.081416] nvme nvme0: allocated 64 MiB host memory buffer.
[    1.096351] nvme nvme0: 4/0/0 default/read/poll queues
[    1.103483] nvme nvme0: Ignoring bogus Namespace Identifiers
[    1.106891]  nvme0n1: p1
[    1.350411] rk_gmac-dwmac fe2a0000.ethernet: IRQ eth_lpi not found
[    1.351383] rk_gmac-dwmac fe2a0000.ethernet: supply phy not found, using dummy regulator
[    1.352245] rk_gmac-dwmac fe2a0000.ethernet: Can not read property: clock_in_out.
[    1.352912] rk_gmac-dwmac fe2a0000.ethernet: Can not read property: tx_delay.
[    1.353542] rk_gmac-dwmac fe2a0000.ethernet: set tx_delay to 0x30
[    1.354083] rk_gmac-dwmac fe2a0000.ethernet: Can not read property: rx_delay.
[    1.354712] rk_gmac-dwmac fe2a0000.ethernet: set rx_delay to 0x10
[    1.355332] rk_gmac-dwmac fe2a0000.ethernet: integrated PHY? (no).
[    1.355919] rk_gmac-dwmac fe2a0000.ethernet: clock input from PHY
[    1.361471] rk_gmac-dwmac fe2a0000.ethernet: init for RGMII_ID
[    1.362284] rk_gmac-dwmac fe2a0000.ethernet: User ID: 0x30, Synopsys ID: 0x51
[    1.362926] rk_gmac-dwmac fe2a0000.ethernet: 	DWMAC4/5
[    1.363387] rk_gmac-dwmac fe2a0000.ethernet: DMA HW capability register supported
[    1.364048] rk_gmac-dwmac fe2a0000.ethernet: RX Checksum Offload Engine supported
[    1.364707] rk_gmac-dwmac fe2a0000.ethernet: TX Checksum insertion supported
[    1.365365] rk_gmac-dwmac fe2a0000.ethernet: Wake-Up On Lan supported
[    1.365936] rk_gmac-dwmac fe2a0000.ethernet: TSO supported
[    1.366424] rk_gmac-dwmac fe2a0000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    1.367132] rk_gmac-dwmac fe2a0000.ethernet: Enabled RFS Flow TC (entries=10)
[    1.367764] rk_gmac-dwmac fe2a0000.ethernet: TSO feature enabled
[    1.368296] rk_gmac-dwmac fe2a0000.ethernet: Using 32/32 bits DMA host/device width
[    1.651622] usbcore: registered new interface driver usb-storage
[    1.652417] i2c_dev: i2c /dev entries driver
[    1.654918] fan53555-regulator 0-001c: FAN53555 Option[12] Rev[15] Detected!
[    1.715058] vdda0v9_image: Bringing 900000uV into 950000-950000uV
[    1.751152] input: rk805 pwrkey as /devices/platform/fdd40000.i2c/i2c-0/0-0020/rk805-pwrkey.2.auto/input/input0
[    1.763201] rk808-rtc rk808-rtc.3.auto: registered as rtc0
[    1.766362] rk808-rtc rk808-rtc.3.auto: setting system clock to 2025-11-12T15:53:31 UTC (1762962811)
[    1.773349] sdhci: Secure Digital Host Controller Interface driver
[    1.773912] sdhci: Copyright(c) Pierre Ossman
[    1.774300] Synopsys Designware Multimedia Card Interface Driver
[    1.775272] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.777979] ledtrig-cpu: registered to indicate activity on CPUs
[    1.779274] scmi_protocol scmi_dev.1: Enabled polling mode TX channel - prot_id:16
[    1.780271] arm-scmi firmware:scmi: SCMI Notifications - Core Enabled.
[    1.780910] arm-scmi firmware:scmi: SCMI Protocol v2.0 'rockchip:' Firmware version 0x0
[    1.784772] NET: Registered PF_INET6 protocol family
[    1.786644] Segment Routing with IPv6
[    1.787013] In-situ OAM (IOAM) with IPv6
[    1.787458] NET: Registered PF_PACKET protocol family
[    1.788109] 8021q: 802.1Q VLAN Support v1.8
[    1.807469] mmc0: SDHCI controller on fe310000.mmc [fe310000.mmc] using ADMA
[    1.838507] dwc3 fcc00000.usb: Configuration mismatch. dr_mode forced to host
[    1.842001] xhci-hcd xhci-hcd.7.auto: xHCI Host Controller
[    1.842519] xhci-hcd xhci-hcd.7.auto: new USB bus registered, assigned bus number 1
[    1.843351] xhci-hcd xhci-hcd.7.auto: hcc params 0x0220fe64 hci version 0x110 quirks 0x0000008002000010
[    1.844238] xhci-hcd xhci-hcd.7.auto: irq 102, io mem 0xfcc00000
[    1.845054] xhci-hcd xhci-hcd.7.auto: xHCI Host Controller
[    1.845556] xhci-hcd xhci-hcd.7.auto: new USB bus registered, assigned bus number 2
[    1.846243] xhci-hcd xhci-hcd.7.auto: Host supports USB 3.0 SuperSpeed
[    1.847577] hub 1-0:1.0: USB hub found
[    1.847976] hub 1-0:1.0: 1 port detected
[    1.848894] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    1.850274] hub 2-0:1.0: USB hub found
[    1.850666] hub 2-0:1.0: 1 port detected
[    1.855193] xhci-hcd xhci-hcd.8.auto: xHCI Host Controller
[    1.855715] xhci-hcd xhci-hcd.8.auto: new USB bus registered, assigned bus number 3
[    1.856541] xhci-hcd xhci-hcd.8.auto: hcc params 0x0220fe64 hci version 0x110 quirks 0x0000008002000010
[    1.857432] xhci-hcd xhci-hcd.8.auto: irq 103, io mem 0xfd000000
[    1.858126] xhci-hcd xhci-hcd.8.auto: xHCI Host Controller
[    1.858632] xhci-hcd xhci-hcd.8.auto: new USB bus registered, assigned bus number 4
[    1.859327] xhci-hcd xhci-hcd.8.auto: Host supports USB 3.0 SuperSpeed
[    1.860674] hub 3-0:1.0: USB hub found
[    1.861092] hub 3-0:1.0: 1 port detected
[    1.862009] usb usb4: We don't know the algorithms for LPM for this host, disabling LPM.
[    1.863401] hub 4-0:1.0: USB hub found
[    1.863795] hub 4-0:1.0: 1 port detected
[    1.865892] clk: Disabling unused clocks
[    1.866073] fan53555-regulator 0-001c: FAN53555 Option[12] Rev[15] Detected!
[    1.867809] dw-apb-uart fe660000.serial: forbid DMA for kernel console
[    1.870063] Waiting for root device PARTUUID=5452574f-02...
[    1.882104] mmc0: new HS200 MMC card at address 0001
[    1.882995] mmcblk0: mmc0:0001 DF4016 14.7 GiB
[    1.884738]  mmcblk0: p1 p2
[    1.885386] mmcblk0boot0: mmc0:0001 DF4016 4.00 MiB
[    1.886448] mmcblk0boot1: mmc0:0001 DF4016 4.00 MiB
[    1.887352] mmcblk0rpmb: mmc0:0001 DF4016 4.00 MiB, chardev (245:0)
[    1.898514] VFS: Mounted root (squashfs filesystem) readonly on device 179:2.
[    1.899837] Freeing unused kernel memory: 2240K
[    1.900294] Run /sbin/init as init process
[    2.001211] init: Console is alive
[    2.001623] init: - watchdog -
[    2.399205] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    2.410762] Key type encrypted registered
[    2.416043] button_hotplug: loading out-of-tree module taints kernel.
[    2.416784] Button Hotplug driver version 0.4.1
[    2.420133] device-mapper: ioctl: 4.48.0-ioctl (2023-03-01) initialised: dm-devel@redhat.com
[    2.431202] r8169 0000:01:00.0: enabling device (0000 -> 0003)
[    2.465400] r8169 0000:01:00.0: can't read MAC address, setting random one
[    2.525571] r8169 0000:01:00.0 eth2: RTL8125B, 22:37:91:e4:51:6e, XID 641, IRQ 104
[    2.526251] r8169 0000:01:00.0 eth2: jumbo features [frames: 16362 bytes, tx checksumming: ko]
[    2.527152] r8169 0001:11:00.0: enabling device (0000 -> 0003)
[    2.555436] r8169 0001:11:00.0: can't read MAC address, setting random one
[    2.602213] r8169 0001:11:00.0 eth3: RTL8125B, d6:bb:ca:c7:70:2f, XID 641, IRQ 105
[    2.602889] r8169 0001:11:00.0 eth3: jumbo features [frames: 16362 bytes, tx checksumming: ko]
[    2.604768] usbcore: registered new interface driver uas
[    2.606100] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    2.617271] init: - preinit -
Cannot parse config file '/etc/fw_env.config': No such file or directory
Failed to find NVMEM device
[    3.174604] rk_gmac-dwmac fe2a0000.ethernet eth1: Register MEM_TYPE_PAGE_POOL RxQ-0
[    3.414858] rk_gmac-dwmac fe2a0000.ethernet eth1: PHY [stmmac-0:01] driver [RTL8211F Gigabit Ethernet] (irq=POLL)
[    3.425873] dwmac4: Master AXI performs any burst length
[    3.426390] rk_gmac-dwmac fe2a0000.ethernet eth1: No Safety Features support found
[    3.427108] rk_gmac-dwmac fe2a0000.ethernet eth1: IEEE 1588-2008 Advanced Timestamp supported
[    3.428185] rk_gmac-dwmac fe2a0000.ethernet eth1: registered PTP clock
[    3.429670] rk_gmac-dwmac fe2a0000.ethernet eth1: configuring for phy/rgmii-id link mode
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    7.781704] mount_root: loading kmods from internal overlay
[    7.795864] kmodloader: loading kernel modules from //etc/modules-boot.d/*
[    7.798949] kmodloader: done loading kernel modules from //etc/modules-boot.d/*
[    7.900579] block: attempting to load /tmp/overlay/upper/etc/config/fstab
[    7.901312] block: unable to load configuration (fstab: Entry not found)
[    7.901974] block: attempting to load /tmp/overlay/etc/config/fstab
[    7.902623] block: unable to load configuration (fstab: Entry not found)
[    7.903288] block: attempting to load /etc/config/fstab
[    7.906779] block: unable to load configuration (fstab: Entry not found)
[    7.907443] block: no usable configuration
[    7.907913] block: attempting to load /etc/config/fstab
[    7.908443] block: unable to load configuration (fstab: Entry not found)
[    7.909072] block: no usable configuration
[    7.909944] loop0: detected capacity change from 0 to 524288
[    7.964867] loop0: detected capacity change from 524288 to 504960
[    8.021622] F2FS-fs (loop0): Mounted with checkpoint version = 561c1a8f
[    8.072496] block: attempting to load /tmp/overlay/upper/etc/config/fstab
[    8.073840] block: extroot: not configured
[    8.074361] block: attempting to load /etc/config/fstab
[    8.074958] block: unable to load configuration (fstab: Entry not found)
[    8.075612] block: no usable configuration
[    8.076916] mount_root: switching to f2fs overlay
[    8.121939] EXT4-fs (mmcblk0p1): mounted filesystem 84173db5-fa99-e35a-95c6-28613cc79ea9 r/w without journal. Quota mode: none.
[    8.127762] EXT4-fs (mmcblk0p1): unmounting filesystem 84173db5-fa99-e35a-95c6-28613cc79ea9.
[    8.130508] urandom-seed: Seeding with /etc/urandom.seed
[    8.190317] procd: - early -
[    8.190674] procd: - watchdog -
[    8.681375] EXT4-fs (nvme0n1p1): mounted filesystem d1712c33-5a26-4221-a208-0bacc20d3511 r/w with ordered data mode. Quota mode: none.
[    8.739305] procd: - watchdog -
[    8.740049] procd: - ubus -
[    8.892850] procd: - init -
Please press Enter to activate this console.
[    9.138075] kmodloader: loading kernel modules from /etc/modules.d/*
[    9.247485] urngd: v1.0.2 started.
[    9.289342] NET: Registered PF_ALG protocol family
[    9.303250] async_tx: api initialized (async)
[    9.305620] raid6: skipped pq benchmark and selected neonx8
[    9.306127] raid6: using neon recovery algorithm
[    9.307892] xor: measuring software checksum speed
[    9.309159]    8regs           :  3931 MB/sec
[    9.310428]    32regs          :  3711 MB/sec
[    9.311636]    arm64_neon      :  3976 MB/sec
[    9.312018] xor: using function: arm64_neon (3976 MB/sec)
[    9.317488] Key type dns_resolver registered
[    9.326220] Key type cifs.idmap registered
[    9.336511] RPC: Registered named UNIX socket transport module.
[    9.337045] RPC: Registered udp transport module.
[    9.337457] RPC: Registered tcp transport module.
[    9.337868] RPC: Registered tcp-with-tls transport module.
[    9.338347] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    9.344838] tun: Universal TUN/TAP device driver, 1.6
[    9.368764] NFS: Registering the id_resolver key type
[    9.369244] Key type id_resolver registered
[    9.369612] Key type id_legacy registered
[    9.375053] ntfs3: Enabled Linux POSIX ACLs support
[    9.378155] fuse: init (API version 7.39)
[    9.386515] usbcore: registered new interface driver lan78xx
[    9.392291] Loading iSCSI transport class v2.0-870.
[    9.394231] usbcore: registered new interface driver ums-alauda
[    9.395254] usbcore: registered new interface driver ums-cypress
[    9.396235] usbcore: registered new interface driver ums-datafab
[    9.397173] usbcore: registered new interface driver ums-freecom
[    9.398121] usbcore: registered new interface driver ums-isd200
[    9.399087] usbcore: registered new interface driver ums-jumpshot
[    9.400034] usbcore: registered new interface driver ums-karma
[    9.401031] usbcore: registered new interface driver ums-sddr09
[    9.401996] usbcore: registered new interface driver ums-sddr55
[    9.403003] usbcore: registered new interface driver ums-usbat
[    9.406046] usbcore: registered new interface driver asix
[    9.407183] usbcore: registered new interface driver ax88179_178a
[    9.408478] usbcore: registered new interface driver cdc_ether
[    9.432534] PPP generic driver version 2.4.2
[    9.433701] NET: Registered PF_PPPOX protocol family
[    9.434995] usbcore: registered new interface driver rndis_host
[    9.436554] iscsi: registered transport (tcp)
[    9.439132] kmodloader: done loading kernel modules from /etc/modules.d/*
[   12.533056] rk_gmac-dwmac fe2a0000.ethernet eth1: Register MEM_TYPE_PAGE_POOL RxQ-0
[   12.774879] rk_gmac-dwmac fe2a0000.ethernet eth1: PHY [stmmac-0:01] driver [RTL8211F Gigabit Ethernet] (irq=POLL)
[   12.784988] dwmac4: Master AXI performs any burst length
[   12.785489] rk_gmac-dwmac fe2a0000.ethernet eth1: No Safety Features support found
[   12.786179] rk_gmac-dwmac fe2a0000.ethernet eth1: IEEE 1588-2008 Advanced Timestamp supported
[   12.787118] rk_gmac-dwmac fe2a0000.ethernet eth1: registered PTP clock
[   12.787767] rk_gmac-dwmac fe2a0000.ethernet eth1: configuring for phy/rgmii-id link mode
[   12.826575] br-lan: port 1(eth1) entered blocking state
[   12.827063] br-lan: port 1(eth1) entered disabled state
[   12.827582] rk_gmac-dwmac fe2a0000.ethernet eth1: entered allmulticast mode
[   12.828474] rk_gmac-dwmac fe2a0000.ethernet eth1: entered promiscuous mode
[   12.895038] Realtek Internal NBASE-T PHY r8169-0-100:00: attached PHY driver (mii_bus:phy_addr=r8169-0-100:00, irq=MAC)
[   13.165389] r8169 0000:01:00.0 eth2: Link is Down
[   13.165848] br-lan: port 2(eth2) entered blocking state
[   13.166332] br-lan: port 2(eth2) entered disabled state
[   13.166835] r8169 0000:01:00.0 eth2: entered allmulticast mode
[   13.167622] r8169 0000:01:00.0 eth2: entered promiscuous mode
[   13.168321] br-lan: port 2(eth2) entered blocking state
[   13.168804] br-lan: port 2(eth2) entered forwarding state
[   13.204884] Realtek Internal NBASE-T PHY r8169-1-1100:00: attached PHY driver (mii_bus:phy_addr=r8169-1-1100:00, irq=MAC)
[   13.485037] r8169 0001:11:00.0 eth3: Link is Down
[   13.486086] br-lan: port 3(eth3) entered blocking state
[   13.486600] br-lan: port 3(eth3) entered disabled state
[   13.487144] r8169 0001:11:00.0 eth3: entered allmulticast mode
[   13.488188] r8169 0001:11:00.0 eth3: entered promiscuous mode
[   13.488969] br-lan: port 3(eth3) entered blocking state
[   13.489464] br-lan: port 3(eth3) entered forwarding state
[   13.498014] rk_gmac-dwmac fe010000.ethernet eth0: Register MEM_TYPE_PAGE_POOL RxQ-0
[   13.734891] rk_gmac-dwmac fe010000.ethernet eth0: PHY [stmmac-1:01] driver [RTL8211F Gigabit Ethernet] (irq=POLL)
[   13.744880] dwmac4: Master AXI performs any burst length
[   13.745400] rk_gmac-dwmac fe010000.ethernet eth0: No Safety Features support found
[   13.746117] rk_gmac-dwmac fe010000.ethernet eth0: IEEE 1588-2008 Advanced Timestamp supported
[   13.747234] rk_gmac-dwmac fe010000.ethernet eth0: registered PTP clock
[   13.748814] rk_gmac-dwmac fe010000.ethernet eth0: configuring for phy/rgmii-id link mode
[   13.786405] br-lan: port 2(eth2) entered disabled state
[   13.787727] br-lan: port 3(eth3) entered disabled state
[   16.407191] r8169 0001:11:00.0 eth3: Link is Up - 2.5Gbps/Full - flow control off
[   16.407939] br-lan: port 3(eth3) entered blocking state
[   16.408445] br-lan: port 3(eth3) entered forwarding state
[   17.938180] rk_gmac-dwmac fe010000.ethernet eth0: Link is Up - 1Gbps/Full - flow control off
[   32.486979] vccio_sd: disabling



BusyBox v1.36.1 (2025-11-12 08:28:48 UTC) built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt 24.10-SNAPSHOT, r0-300ebaf
 -----------------------------------------------------
=== WARNING! =====================================
There is no root password defined on this device!
Use the "passwd" command to set up a new password
in order to prevent unauthorized SSH logins.
--------------------------------------------------
```

However, I encountered a GMAC initialization failure during one of the cold boots (It returned to normal after restarting). Logs shows that `mdio_bus stmmac-1: MDIO device at address 1 is missing.`:

```
[    0.492655] random: crng init done
[    0.498185] loop: module loaded
[    0.498490] mtip32xx Version 1.3.1
[    0.500332] nvme nvme0: pci function 0002:21:00.0
[    0.500788] nvme 0002:21:00.0: enabling device (0000 -> 0002)
[    0.504448] rk_gmac-dwmac fe010000.ethernet: IRQ eth_lpi not found
[    0.505201] rk_gmac-dwmac fe010000.ethernet: supply phy not found, using dummy regulator
[    0.506079] rk_gmac-dwmac fe010000.ethernet: Can not read property: clock_in_out.
[    0.506746] rk_gmac-dwmac fe010000.ethernet: Can not read property: tx_delay.
[    0.507377] rk_gmac-dwmac fe010000.ethernet: set tx_delay to 0x30
[    0.507919] rk_gmac-dwmac fe010000.ethernet: Can not read property: rx_delay.
[    0.508547] rk_gmac-dwmac fe010000.ethernet: set rx_delay to 0x10
[    0.509097] rk_gmac-dwmac fe010000.ethernet: integrated PHY? (no).
[    0.509682] rk_gmac-dwmac fe010000.ethernet: clock input from PHY
[    0.512658] hwmon hwmon0: temp1_input not attached to any thermal zone
[    0.515237] rk_gmac-dwmac fe010000.ethernet: init for RGMII_ID
[    0.516053] rk_gmac-dwmac fe010000.ethernet: User ID: 0x30, Synopsys ID: 0x51
[    0.516718] rk_gmac-dwmac fe010000.ethernet: 	DWMAC4/5
[    0.517181] rk_gmac-dwmac fe010000.ethernet: DMA HW capability register supported
[    0.517844] rk_gmac-dwmac fe010000.ethernet: RX Checksum Offload Engine supported
[    0.518504] rk_gmac-dwmac fe010000.ethernet: TX Checksum insertion supported
[    0.519126] rk_gmac-dwmac fe010000.ethernet: Wake-Up On Lan supported
[    0.519696] rk_gmac-dwmac fe010000.ethernet: TSO supported
[    0.520182] rk_gmac-dwmac fe010000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    0.520888] rk_gmac-dwmac fe010000.ethernet: Enabled RFS Flow TC (entries=10)
[    0.521521] rk_gmac-dwmac fe010000.ethernet: TSO feature enabled
[    0.522054] rk_gmac-dwmac fe010000.ethernet: Using 32/32 bits DMA host/device width
[    0.523623] mdio_bus stmmac-1: MDIO device at address 1 is missing.
[    0.526788] rk_gmac-dwmac fe2a0000.ethernet: IRQ eth_lpi not found
[    0.527737] rk_gmac-dwmac fe2a0000.ethernet: supply phy not found, using dummy regulator
[    0.528602] rk_gmac-dwmac fe2a0000.ethernet: Can not read property: clock_in_out.
[    0.529270] rk_gmac-dwmac fe2a0000.ethernet: Can not read property: tx_delay.
[    0.529902] rk_gmac-dwmac fe2a0000.ethernet: set tx_delay to 0x30
[    0.530444] rk_gmac-dwmac fe2a0000.ethernet: Can not read property: rx_delay.
[    0.531073] rk_gmac-dwmac fe2a0000.ethernet: set rx_delay to 0x10
[    0.531623] rk_gmac-dwmac fe2a0000.ethernet: integrated PHY? (no).
[    0.532209] rk_gmac-dwmac fe2a0000.ethernet: clock input from PHY
[    0.537759] rk_gmac-dwmac fe2a0000.ethernet: init for RGMII_ID
[    0.538570] rk_gmac-dwmac fe2a0000.ethernet: User ID: 0x30, Synopsys ID: 0x51
[    0.539217] rk_gmac-dwmac fe2a0000.ethernet: 	DWMAC4/5
[    0.539680] rk_gmac-dwmac fe2a0000.ethernet: DMA HW capability register supported
[    0.540340] rk_gmac-dwmac fe2a0000.ethernet: RX Checksum Offload Engine supported
[    0.540998] rk_gmac-dwmac fe2a0000.ethernet: TX Checksum insertion supported
[    0.541618] rk_gmac-dwmac fe2a0000.ethernet: Wake-Up On Lan supported
[    0.542184] rk_gmac-dwmac fe2a0000.ethernet: TSO supported
[    0.542670] rk_gmac-dwmac fe2a0000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    0.543376] rk_gmac-dwmac fe2a0000.ethernet: Enabled RFS Flow TC (entries=10)
[    0.544009] rk_gmac-dwmac fe2a0000.ethernet: TSO feature enabled
[    0.544589] rk_gmac-dwmac fe2a0000.ethernet: Using 32/32 bits DMA host/device width
```

1715173329 commented:
The kernel now more or less relies on u-boot to reset the PHY first. You may probably want to create a dedicated u-boot for this board instead of using the generic-rk3568 configuration.

jjm2473 commented:
@1715173329 
I found that moving `reset-*` from `rgmii_phy` to `snps,reset-*` in `gmac` solves this problem, but `snps,reset-*` is deprecated by upstream linux.

dts for R1 is expected to be added in Linux 6.19, I think U-Boot may not accept R1 patches right now.

So should I use `snps,reset-*` as a temporary workaround, and then switch to a dedicated U-Boot once it's ready? Or do I have to wait until the upstream U-Boot is ready?

1715173329 commented:
> I found that moving `reset-*` from `rgmii_phy` to `snps,reset-*` in `gmac` solves this problem, but `snps,reset-*` is deprecated by upstream linux.
> 

Yes.

> dts for R1 is expected to be added in Linux 6.19, I think U-Boot may not accept R1 patches right now.

Just create a patch for uboot-rockchip for now, it's okay.

> 
> So should I use `snps,reset-*` as a temporary workaround, and then switch to a dedicated U-Boot once it's ready? Or do I have to wait until the upstream U-Boot is ready?

jjm2473 committed:
uboot-rockchip: add dedicated uboot for LinkEase EasePi R1

The generic-rk3568 does not reset the gmac phy, causing the kernel to
fail to initialize gmac during cold boot with the error message
"mdio_bus stmmac-1: MDIO device at address 1 is missing."

Adding a dedicated uboot for EasePi R1 to resolve this issue.

The EasePi R1's DTS hasn't been merged into the Linux release yet,
so it cannot be submitted to the upstream u-boot.

Fixes: ef093299db ("uboot-rockchip: add support for LinkEase EasePi R1")
Signed-off-by: Liangbin Lian <jjm2473@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20772
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit 5dff7b45d30f7aea5c7d10f276ce4c72ec6bd6ed)

With changes:
1. Reorder patch file names (`105-*` to `106-*`).
2. configs/easepi-r1-rk3568_defconfig needs CONFIG_SPL_PAD_TO on u-boot 2024.10.

Signed-off-by: Liangbin Lian <jjm2473@gmail.com>

jjm2473 committed:
rockchip: LinkEase EasePi R1 use dedicated uboot

Fixes the "mdio_bus stmmac-1: MDIO device at address 1 is missing" error
that occurs during cold boot when initializing GMAC.

Fixes: 8ca4caacd039 ("rockchip: Add support for RK3568 LinkEase EasePi R1")
Signed-off-by: Liangbin Lian <jjm2473@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20772
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit e6a5ca0da7377645be7b9ba70b18de3b9bab5342)

jjm2473 commented:
main branch PR for uboot-rockchip is ready for review https://github.com/openwrt/openwrt/pull/20772

jjm2473 committed:
rockchip: make NIC name predictable for LinkEase EasePi R1

The probe order for PCIe buses and devices is non-deterministic,
making the names eth2 and eth3 unpredictable (they may be swapped).

This patch fixes the names by referencing the device path using
`ucidef_set_network_device_path`.

This patch ensures that the OpenWrt interface name matches the case label.

Fixes: 8ca4caacd039 ("rockchip: Add support for RK3568 LinkEase EasePi R1")
Signed-off-by: Liangbin Lian <jjm2473@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20779
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit d7530c1e5405cf93ecd0a38ef758cba5760dfd97)

jjm2473 commented:
ping @hauke

hauke commented:
Applied to openwrt-24.10 in e35f37791f, c795a91579, fbaf5271d2, 9c25d39d3e and 414f196a96.
